Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ISISIndirectEnergyTransferWrapper.py

```diff
@@ -0,0 +1,200 @@
+        self._calibration_workspace = workspace_or_none(self.getPropertyValue('CalibrationWorkspace'))
+
+        self._instrument_name = self.getPropertyValue('Instrument')
+        self._analyser = self.getPropertyValue('Analyser')
+        self._reflection = self.getPropertyValue('Reflection')
+        self._efixed = self.getProperty('Efixed').value
+
+        self._spectra_range = self.getProperty('SpectraRange').value
+        self._background_range = elements_or_none(self.getProperty('BackgroundRange').value)
+        self._rebin_string = string_or_none(self.getPropertyValue('RebinString'))
+        self._detailed_balance = self.getProperty('DetailedBalance').value
+        self._scale_factor = self.getProperty('ScaleFactor').value
+        self._fold_multiple_frames = self.getProperty('FoldMultipleFrames').value
+
+        self._grouping_method = self.getPropertyValue('GroupingMethod')
+        self._grouping_workspace = workspace_or_none(self.getPropertyValue('GroupingWorkspace'))
+        self._grouping_string = string_or_none(self.getPropertyValue('GroupingString'))
+        self._grouping_map_file = string_or_none(self.getPropertyValue('MapFile'))
+
+        self._output_x_units = self.getPropertyValue('UnitX')
+        self._output_workspace = self.getPropertyValue('OutputWorkspace')
+
+    def PyExec(self):
+        self._setup()
+        self._reduce_data()
+
+        self.setProperty('OutputWorkspace', get_ads_workspace(self._output_workspace))
+
+    def _reduce_data(self):
+        reduction_algorithm = self.createChildAlgorithm(name='ISISIndirectEnergyTransfer', startProgress=0.1,
+                                                        endProgress=1.0, enableLogging=False)
+        reduction_algorithm.enableHistoryRecordingForChild(False)
+
+        args = {"InputFiles": self._data_files, "SumFiles": self._sum_files, "LoadLogFiles": self._sum_files,
+                "CalibrationWorkspace": self._calibration_workspace, "Instrument": self._instrument_name,
+                "Analyser": self._analyser, "Reflection": self._reflection, "SpectraRange": self._spectra_range,
+                "BackgroundRange": self._background_range, "RebinString": self._rebin_string,
+                "ScaleFactor": self._scale_factor, "FoldMultipleFrames": self._fold_multiple_frames,
+                "GroupingMethod": self._grouping_method, "GroupingWorkspace": self._grouping_workspace,
+                "GroupingString": self._grouping_string, "MapFile": self._grouping_map_file,
+                "UnitX": self._output_x_units, "OutputWorkspace": self._output_workspace}
+
+        if self._efixed != Property.EMPTY_DBL:
+            args["Efixed"] = self._efixed
+
+        if self._detailed_balance != Property.EMPTY_DBL:
+            args["DetailedBalance"] = self._detailed_balance
+
+        for key, value in args.items():
+            if value is not None:
+                reduction_algorithm.setProperty(key, value)
```

Comment:
check with @Mantid-Matthew but I believe there is a way to do all of the args at once

Reply:
you can do alg.setProperties(parameter_dict) although this won't replicate the checking the values are not None if that is important here.

Reply:
This will be useful in the future as I didn't know you could do this, but in this case the check that the values are not None is important